polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1050,6 +1050,47 @@ async fn second_unblocked_collations<Context>(
 	}
 }
 
+fn ensure_seconding_limit_is_respected(
+	relay_parent: &Hash,
+	para_id: ParaId,
+	state: &State,
+) -> std::result::Result<(), AdvertisementError> {
+	let paths = state.implicit_view.paths_via_relay_parent(relay_parent);
+
+	gum::trace!(
+		target: LOG_TARGET,
+		?relay_parent,
+		?para_id,
+		?paths,
+		"Checking seconding limit",
+	);
+
+	for path in paths {
```

Comment:
I've added a todo in https://github.com/paritytech/polkadot-sdk/issues/6679 to keep `ClaimqueueState` in `State` and have it ready for use when we needed instead of rebuilding it each time.